### PR TITLE
[REVIEW] libcuspatial benchmark scaffolding

### PR DESCRIPTION
**Pull request by cwharris**
#168

Basic copy-pasta and rejiggering from cudf/rmm benchmark scaffolding.

May want to use cudf's `benchmark_fixture.hpp` and rmm's `synchonrization.hpp` directly, rather than maintain our own copy?

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by harrism** [comment]
> May want to use cudf's `benchmark_fixture.hpp` and rmm's `synchonrization.hpp` directly, rather than maintain our own copy?

Why not get both from the same repo?

File: cpp/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+project(CUSPATIAL_BENCHS LANGUAGES C CXX CUDA)
```
Comment (harrism) [inline cpp/benchmarks/CMakeLists.txt:3]:
```suggestion
project(CUSPATIAL_BENCHMARKS LANGUAGES C CXX CUDA)
```

(or at least spell BENCHES correctly. :) )

File: cpp/benchmarks/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+find_library(CUDFTESTUTIL_LIBRARY "libcudftestutil.a"
+             HINTS "$ENV{CUDF_ROOT}/lib")
+
+message(STATUS "CUDFTESTUTIL: CUDFTESTUTIL_LIBRARY set to ${CUDFTESTUTIL_LIBRARY}")
+message(STATUS "CUDFTESTUTIL: CUDF_TEST_INCLUDE set to ${CUDF_TEST_INCLUDE}")
+
+add_library(cudftestutil STATIC IMPORTED ${CUDFTESTUTIL_LIBRARY})
+if (CUDF_TEST_INCLUDE AND CUDFTESTUTIL_LIBRARY)
+    set_target_properties(cudftestutil PROPERTIES IMPORTED_LOCATION ${CUDFTESTUTIL_LIBRARY})
```
Comment (harrism) [inline cpp/benchmarks/CMakeLists.txt:20]: Might need position independent code option here (@trxcllnt  did that work for you?)

Comment (trxcllnt) [reply cpp/benchmarks/CMakeLists.txt:20]: It did not. I couldn't replicate with gcc 7.5 in rapids-compose, but I think CI might be using gcc 7.3.

File: cpp/benchmarks/CMakeLists.txt (hunk 1 above)
Comment (cwharris) [reply cpp/benchmarks/CMakeLists.txt:3]: Consistency! :laughing: 

https://github.com/rapidsai/cudf/blob/1046eed3b8a59792fa6ed5f66a34cc5b792ef929/cpp/benchmarks/CMakeLists.txt#L3

File: cpp/benchmarks/CMakeLists.txt (hunk 2 above)
Comment (cwharris) [reply cpp/benchmarks/CMakeLists.txt:20]: Going to keep this as-is until we can determine what change is needed, if any.

**Pull request merged**